sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -610,6 +610,23 @@ class CudfRegexTranspiler(replace: Boolean) {
           // example: "a*+"
           throw new RegexUnsupportedException(nothingToRepeat)
 
+        case (RegexGroup(capture, term), SimpleQuantifier(ch)) if "+*".contains(ch) =>
+          // example: "(3?)+"
+          def isSimpleRepetition(e: RegexAST):Boolean = {
+            e match {
+              case RegexRepetition(term, quantifier) =>
+                term.isInstanceOf[RegexCharacterClassComponent]
```

Comment:
We can use pattern matching more fully here and avoid the `isInstanceOf`. Also, `quantifier` is unused here.

```suggestion
              case RegexRepetition(_: RegexCharacterClassComponent, _) => true
```